### PR TITLE
LoRa Module (semtech-baed)

### PR DESCRIPTION
**Pull request by quartzjer**
This is the start of a module to support the [Semtech SX1276](https://www.semtech.com/wireless-rf/rf-transceivers/sx1276/) based radios.  It's a work-in-progress!

### CODE REVIEW

**Comment by matthijskooijman** [comment]
The problem with this code is, I think, that calling the various receive functions in the SX1272 library will switch the LoRa module into RX mode, wait for a packet to be received within the given timeout period. I _think_ it doesn't actually switch back to standby mode after reception (except when it succesfully received a packet that was for someone else?). However, I think that the next call to a receive function might mess up any pending reception.

In other words, this library is really intended for one-off receives with longer timeouts, while we  would need a more asynchronous polling type of approach instead.

Given that I don't really like the library code itself (lots of similar functions, some magic constants, LGPL license), I'll have a look at the LMIC code instead.

**Comment by quartzjer** [comment]
:+1: it was just used as the fastest way to validate the hardware was hooked up right, you're the only analysis so far :)
